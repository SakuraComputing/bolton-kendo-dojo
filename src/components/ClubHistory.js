import React from 'react'
import { connect } from 'react-redux';
import Spinner from '../common/Spinner';

export class ClubHistory extends React.Component {

  render() {

    let clubHistory;

    const { club, loading } = this.props.club;

    if(club !== null && !loading) {
      clubHistory = (
        <div>{club[0].clubHistory}</div>
      )
    } else {
      clubHistory = <Spinner />
    }

    return (
      <div>
        <h1>Club History</h1>
        {clubHistory}
      </div>
    )
  }
}
const mapStateToProps = (state) => ({
  club: state.club
})
export default connect(mapStateToProps)(ClubHistory);
