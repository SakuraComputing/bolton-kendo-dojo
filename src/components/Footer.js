import React from 'react'

const Footer = () => {
  return (
    <div className="footer">
      <div className="bka-details">
        <a className="bka-link" href="http://www.britishkendoassociation.com" target="_blank" rel="noopener noreferrer">Member of the British Kendo Association<img src="/img/bka_logo3.png" alt="BKA Logo" className="bka-logo"/></a>
      </div>
      <div className="copyright">
        Copyright &copy; {new Date().getFullYear()} Bolton Kendo Club
      </div>
    </div>
  )
}
export default Footer;
