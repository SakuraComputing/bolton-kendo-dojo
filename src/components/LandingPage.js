import React from 'react'

const LandingPage = () => {
  return (
    <div>
        <h1>Bolton Kendo Dojo</h1>
    </div>
  )
}
export default LandingPage;
