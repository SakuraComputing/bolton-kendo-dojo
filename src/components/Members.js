import React from 'react';
import { Link } from 'react-router-dom';

const  Members = () => {
  return (
    <div className="members">
      <h1 className="members-header">Welcome to the Members Area</h1>
      <div className="members-area-container">
        <div className="members-area-link">
          <Link to="/memberphoto">Photographs</Link>
          <p>Select the photgraphs to go in to the photgraph section</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Kendo Mind Map</Link>
          <p>Enter the Kendo Mind Map section for all deeper kendo study</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Videos</Link>
          <p>Go into the video section and view private members videos</p>
        </div>
        <div className="members-area-link">
          <Link to="/members">Bulletin Board</Link>
          <p>Add a post to the bulletin board. Responds to other posts</p>
        </div>
      </div>
    </div>
  )
}
export default Members;
