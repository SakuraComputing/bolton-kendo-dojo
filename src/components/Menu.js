import React from 'react'
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { logoutUser } from '../actions/authActions';

export class Menu extends React.Component {

  onLogoutClick = (e) => {
      e.preventDefault();
      this.props.logoutUser();
  }  

  render() { 

        const { isAuthenticated } = this.props.auth;

        const authLinks = (
            <div>
                <li>
                    <Link to="/study">Kendo Study</Link>
                </li>
                <li>
                    <Link
                        to="/members"
                    >
                    Members
                    </Link>
                </li>
                <li>
                    <Link 
                        to="/logout"
                        href=""
                        onClick={this.onLogoutClick}
                    >
                    Logout
                    </Link>
                </li>
            </div>
        )

        const guestLinks = (
            <li>
                <Link to="/login">Login / Registration</Link>
            </li>
        )

        return (
            <div>
                <ul className="nav">
                    <li>
                        <Link to="/">Home</Link>
                    </li>
                    <li>
                        <Link to="/about">Where and When</Link>
                    </li>
                    <li>
                        <Link to="/history">Club History</Link>
                    </li>
                    <li>
                        <Link to="/theory">What is Kendo</Link>
                    </li>
                    <li>
                        <Link to="/links">Kendo Links</Link>
                    </li>
                    <li>
                        <Link to="/photos">Kendo Photograph's</Link>
                    </li>
                    <li>
                        <Link to="/videos">Kendo Videos</Link>
                    </li>
                    {isAuthenticated ? authLinks :  guestLinks}
                </ul>
            </div>
        )
    }
}
const mapStateToProps = state => ({
    auth: state.auth
})
export default connect(mapStateToProps, { logoutUser})(Menu);
