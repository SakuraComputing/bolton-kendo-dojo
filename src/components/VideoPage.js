import React, { Component } from 'react';
import { connect } from 'react-redux';
import Spinner from '../common/Spinner';
// import ReactPlayer from 'react-player';

export class VideoPage extends Component {

    render() {

        let video;

        const { club, loading } = this.props.club;
        
        if(club !== null && !loading) {
            video = club[0].videos.map(((video, key) => 
                <div key={key}>{video.videoId}</div>
            ))
        } else {
            video = <Spinner />
        }

        return (
            <div className="content-container">
                {video}
            </div>
        )
    }
}

const mapStateToProps = state => ({
    club: state.club
})
export default connect(mapStateToProps, null)(VideoPage);
