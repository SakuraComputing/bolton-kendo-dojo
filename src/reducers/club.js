import { GET_CLUB_PROFILE } from '../actions/types';

const initialState = {
    loading: false,
    club: {}
};

export default (state = initialState, action) => {
    switch(action.type) {
        case GET_CLUB_PROFILE: 
            return {
                ...state,
                club: action.payload,
                loading: false
            }
        default:
            return state
    }

}