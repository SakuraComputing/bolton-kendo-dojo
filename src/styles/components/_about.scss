.about-header {
    text-align: center;
}

.about-map {
   margin: 5%;
}

.about-map-section {
    border: solid 1px $blue-black;
    border-radius: 5px;-bottom: 1em;
    margin-bottom: 20px;
}

.about-map-section:hover {
    background-color: rgb(220, 235, 243);
}
 

.about-container {
    margin: 8%;
    background-color: rgb(244, 246, 250);
}

.about-section {
    border: solid 1px $blue-black;
    border-radius: 5px;
    margin-bottom: 7px;
    text-align: center;
}

.about-section:hover {
    background-color: rgb(220, 235, 243);
}