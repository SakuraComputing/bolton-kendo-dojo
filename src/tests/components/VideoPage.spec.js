import React from 'react';
import { shallow } from 'enzyme';
import { VideoPage } from '../../components/VideoPage';
import club from '../fixtures/club'; 

describe('Video Page Component', () => {

    let videoPage;

    beforeEach(() => {
        videoPage = shallow(<VideoPage club={{club}} />)
    });
    it('should render correctly', () => {
        expect(videoPage).toMatchSnapshot();
    });

    it('should not display a spinner when state loading is false', () => {
        const props = {
            club: {
                loading: false,
                club
            },
        }
        videoPage = shallow(<VideoPage {...props} />)
        expect(videoPage.find('Spinner').length).toEqual(0);
    });

    it('should display a spinner when state loading is true', () => {
        const props = {
            club: {
                loading: true,
                club: {}
            },
        }
        videoPage = shallow(<VideoPage {...props} />)
        expect(videoPage.find('Spinner').length).toEqual(1);
    });


});