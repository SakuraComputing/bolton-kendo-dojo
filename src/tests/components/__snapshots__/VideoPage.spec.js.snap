// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Page Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPage
    club={
      Object {
        "club": Array [
          Object {
            "__v": 9,
            "_id": "5b96fdds7463ea1d37be8751b7",
            "about": "Some History",
            "addresses": Array [
              Object {
                "_id": "5b9egfd516aa75c12041f5c6b",
                "address1": "School",
                "address2": "Church Street",
                "address3": "Anywhere",
                "address4": "Trumpton",
                "postcode": "TR35YG",
              },
            ],
            "clubHistory": "History of the club 
",
            "clubName": "Test Club",
            "contacts": Array [
              Object {
                "_id": "5b9eb4ca222bdb0dfcec0a72",
                "email": "tbobe@dr.com",
                "name": "T Bone Walker",
                "telephoneNumber": "764554 453457",
              },
            ],
            "practices": Array [
              Object {
                "_id": "5b9ebf390e03gfgfdg0fe4db8544",
                "time": "Sunday 10:00am - 12:00pm (2 Hours)",
                "venue": "Room",
              },
            ],
            "videos": Array [
              Object {
                "videoId": "8Ea2ih7uNW4",
              },
              Object {
                "videoId": "oZ3M3mbWZ9A",
              },
              Object {
                "videoId": "Q4SHWXQBVL4",
              },
            ],
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          8Ea2ih7uNW4
        </div>,
        <div>
          oZ3M3mbWZ9A
        </div>,
        <div>
          Q4SHWXQBVL4
        </div>,
      ],
      "className": "content-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "8Ea2ih7uNW4",
        },
        "ref": null,
        "rendered": "8Ea2ih7uNW4",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "oZ3M3mbWZ9A",
        },
        "ref": null,
        "rendered": "oZ3M3mbWZ9A",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Q4SHWXQBVL4",
        },
        "ref": null,
        "rendered": "Q4SHWXQBVL4",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            8Ea2ih7uNW4
          </div>,
          <div>
            oZ3M3mbWZ9A
          </div>,
          <div>
            Q4SHWXQBVL4
          </div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "8Ea2ih7uNW4",
          },
          "ref": null,
          "rendered": "8Ea2ih7uNW4",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "oZ3M3mbWZ9A",
          },
          "ref": null,
          "rendered": "oZ3M3mbWZ9A",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Q4SHWXQBVL4",
          },
          "ref": null,
          "rendered": "Q4SHWXQBVL4",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
